apps/src/code-studio/assets/DefaultSpritesEditor.jsx

```diff
@@ -30,11 +31,23 @@ export default class DefaultSpritesEditor extends React.Component {
       });
   }
 
+  incrementPendingChanges() {
+    let changes = this.state.pendingChangesCount + 1;
+    this.setState({pendingChangesCount: changes});
```

Comment:
now that we have two operations that can write this state, i think we should make it idempotent. right now, there's technically a race condition (a very unlikely one) because we read from state, then write the state, which could have changed between the two lines if `incrementPendingChanges` were called at basically the same time.

we can make this idempotent by using the state given within the `setState` callback:

```javascript
this.setState(state => {
  pendingChangesCount: state.pendingChangesCount + 1
});
```

this is mostly a teaching moment 😄 not a race condition that would likely happen, but it's still useful to know how to handle this case!